### PR TITLE
Update particles.lua

### PR DESCRIPTION
**Pull request by FabberXD**
Added rotation, start_scale, end_scale properties
Reworked life system

### CODE REVIEW

File: lua/modules/particles.lua [hunk 1]
```diff
@@ -60,20 +65,33 @@ particles.newEmitter = function(_, config)
...
-				table.insert(particles.pool, p)
-			end)
+			p.Tick = function(s, dt)
+				s.Scale = particles.lerp(s.startScale, s.endScale, 1-(s.life/s.startLife))
+				
+				s.life = s.life - dt
+				if s.life <= 0 then
+					s:remove()
+				end
+			end
```
Comment (aduermael) [inline lua/modules/particles.lua:94]: It's not ideal to instantiate one Tick for each particle. 
Actually, Timers weren't ideal even. 
Now that we have local events, the module should listen for tick events. 
And each particle could carry its creation date `Time:UnixMilli()`. 
Within one single tick, we could compare with `now` to learn and remove when particle's life is over.

File: lua/modules/particles.lua [hunk 2]
```diff
@@ -60,20 +65,33 @@ particles.newEmitter = function(_, config)
...
+			p.Rotation = particles.fromConfigOrDefault(conf.rotation, Rotation(0, 0, 0))
 			p.Acceleration = particles.fromConfigOrDefault(conf.acceleration, Number3(0, 0, 0))
-			p.Scale = particles.fromConfigOrDefault(conf.scale, 1.0)
+			p.startScale = particles.fromConfigOrDefault(conf.start_scale, particles.fromConfigOrDefault(conf.scale, 1.0))
+			p.endScale = particles.fromConfigOrDefault(conf.end_scale, particles.fromConfigOrDefault(conf.start_scale, particles.fromConfigOrDefault(conf.scale, 1.0)))
 			p.Mass = particles.fromConfigOrDefault(conf.mass, 1.0)
 			p.Palette[1].Color = particles.fromConfigOrDefault(conf.color, Color.White)
 			p.CollisionGroups = particles.fromConfigOrDefault(conf.collisionGroups, nil)
 			p.CollidesWithGroups = particles.fromConfigOrDefault(conf.collidesWithGroups, nil)
+			p.startLife = particles.fromConfigOrDefault(conf.life, 10.0)
```
Comment (aduermael) [inline lua/modules/particles.lua:76]: Considering my comment about Tick, it's not necessary to add this `startLife` field.

**Review by aduermael — CHANGES_REQUESTED** [review]

File: lua/modules/particles.lua [hunk 3]
```diff
@@ -16,6 +16,9 @@ local particles = {
 			return default
 		end
 	end,
+	lerp = function(a, b, w)
+		return a + (b-a)*w
+	end
```
[the remark this answers is not in the document]
Comment (FabberXD) [reply lua/modules/particles.lua:21]: ok! will update this

File: lua/modules/particles.lua (hunk 1 above)
Comment (FabberXD) [reply lua/modules/particles.lua:91]: i think it will be worse that this system

**Pull request closed without merging**